### PR TITLE
RESTORE THE CUBE

### PR DESCRIPTION
**Pull request by brimatteng**
image

 Add the styles, html, and inline css for the CUBE and add a new font while we are at it!

I like the idea of a crunchy gothic header motif, this font is free-ware so it's
a-okay for us to use regardless. It's only a partial set of mostly letters (not
even sure there are numbers here )

This change:
- Adds the new font face "bu Gothic Hybrid" to the new dir `/app/assets/fonts` with both the "woff" and
  "woff2" formats. (This is a web font thing, if you're going to do local you
  might want to use the "ttf" or "otf" formats which I have but which are not
  included. This also includes a fonts.css under `app/assets/stylesheets` to
  load the font automatically

- Add that new font to the sprockets powered asset pipeline, this adds
  fingerprinting, gziping, and precompiling support without us having to do
  anything or reach for something like esbuild, broccoli, or webpacker. (True
  story: I have no idea how to use any of those so I'm sticking to rails
  defaults until proven otherwise)

- Import our new font assets to the whole of the application using a
  stylesheet_link_tag

- add an extension to our tailwind.config.js to enable us to use a
  "font-gothic" html class name to automatically get our gothic font set. (See
  the home_page.html.erb for context)

- Turn on source maps in development, this is pretty much just for debugging and
  better tooling in the browser inspector

- Add the `Rack::Deflator` middleware. This is built in support for gzipping data sent over the wire. 
  The TLDR is that it's going to be turned on globally and always gzip requests. The idea is CPUs
  are always faster than the network IO so send less data and make the CPU do the lifting. 
  The real benefits will be on phones with shit service.

### CODE REVIEW

**Comment by brimatteng** [comment]
https://github.com/brimatteng/magister/projects/1#card-79340463

**Review by davidneto92 — COMMENTED** [review]
Looks pretty deece to me
